### ISSUE TITLE
refactor(proposal): lambas over regular loops

### ISSUE THREAD

**Issue by mikededo** [github-stats-22/github-stats#104]
### Description

After discussing with @plozanol about which syntax should we use, in order to have a more unified code style, we have concluded that we prefer the lamba function syntax (see the [trigger of the discussion](https://github.com/github-stats-22/github-stats/pull/102#issuecomment-1348374251)).

Therefore, since #70 issue has been initially developed using regular for loops (because our little knowledge about the language), and if this decision is accepted by the rest of the team, I would refactor the code added due to #70 in order to use lambdas.

**Comment by sdomingobasora** [comment]
OK for me!

**Comment by mikededo** [comment]
Oh wow, that's eye candy me. I will try to find a similar solution to how it is being implemented. 🥳

**Comment by mikededo** [comment]
I just realised that yesterday's evening I was trying to implement it using lambdas but, since lambdas create a new scope, the we would have to even nest mor `try-catch` inside each lambda.

Let me explain a bit more. Inside each iteration, we are running one or many use cases that implement one of the external repositories. Such interfaces, contain one or multiple methods, each of them throwing an `HttpException`. If we use lambdas for our code, that can be translated into creating an inline-function, therefore we would have to take care of the exception. From what I investigated, there's no such way of defining the `throws` keyword on the lambda declaration. This would imply the use of a `try-catch`, which would add more nesting into the code.

Solutions that I have come up with:

- Keeping the code as it is, using for with the `for-each` notation.
- Try to find if it is possible to refactor the code into functions and make this new functions the actual lambda.

Open to thoughts. I'm biased towards the second option.
